eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/GA4GHBeaconWSServerV2.java

```diff
@@ -0,0 +1,325 @@
+
+        return queryGet(requestBody.getReferenceName(),
+                requestBody.getStart(),
+                requestBody.getStartMin(),
+                requestBody.getStartMax(),
+                requestBody.getEnd(),
+                requestBody.getEndMin(),
+                requestBody.getEndMax(),
+                requestBody.getReferenceBases(),
+                requestBody.getAlternateBases(),
+                requestBody.getVariantType(),
+                requestBody.getAssemblyId(),
+                requestBody.getDatasetIds(),
+                requestBody.getIncludeDatasetResponses(),
+                response);
+    }
+
+    public String checkErrorHelper(BeaconAlleleRequestBody request) {
+
+        if (request.getStart() != null && request.getStart() < 0) {
+            return "please provide a positive start number";
+        }
+
+        if (request.getEnd() != null && request.getEnd() < 0) {
+            return "pleaseprovide a positive end number";
+        }
+
+        if (request.getAlternateBases() == null && request.getVariantType() == null) {
+            return "Either alternateBases ot variantType is required";
+        }
+
+        return null;
+    }
+
+    public List<DatasetAlleleResponse> getDatasetAlleleResponsesHelper(List<VariantMongo> variantMongoList,
+                                                                       BeaconAlleleRequestBody request) {
+
+        List<DatasetAlleleResponse> datasetAllelResponses = new ArrayList<DatasetAlleleResponse>();
+
+        if (request.getIncludeDatasetResponses() == null ||
+                request.getIncludeDatasetResponses().equalsIgnoreCase("NONE")) {
+            return null;
+        }
+
+        List<VariantSourceMongo> variantSourceMongoList = service.findAllForBeacon();
+
+        HashSet<String> studiesPresent = new HashSet<String>();
+        HashMap<String, Float> studyIdToFrequencyMapper = new HashMap<>();
+        variantMongoList.forEach(variantMongo -> {
+            variantMongo.getSourceEntries().forEach(variantSourceEntryMongo -> {
+                studiesPresent.add(variantSourceEntryMongo.getStudyId());
+            });
+            variantMongo.getVariantStatsMongo().forEach(variantStatisticsMongo -> {
+                if(variantMongo.getAlternate().equalsIgnoreCase(variantStatisticsMongo.getMafAllele())) {
+                    studyIdToFrequencyMapper.put(variantStatisticsMongo.getStudyId(), variantStatisticsMongo.getMaf());
+                }
+            });
+        });
+
+        HashMap<String, VariantSourceMongo> allStudies = new HashMap<>();
+        variantSourceMongoList.forEach(variantSourceMongo -> {
+            allStudies.put(variantSourceMongo.getStudyId(), variantSourceMongo);
+        });
+
+        if (request.getIncludeDatasetResponses().equalsIgnoreCase("ALL")) {
+            allStudies.forEach((studyId, variantSourceMongo) -> {
+                if (studiesPresent != null && studiesPresent.contains(studyId)) {
```

Comment:
you instanciated `studiesPresent` in this method. no need to check for null

Reply:
Okay, I will fix it

Reply:
Fixed :+1: